packages/sdk/src/inscription/witness.ts

```diff
@@ -68,16 +59,24 @@ export function buildWitnessScript({ recover = false, ...options }: WitnessScrip
   ])
 }
 
-function opPush(str: string, encoding: BufferEncoding = "utf8") {
-  const buff = Buffer.from(str, encoding)
-  const obj = [buff]
-  const push = Buffer.concat(obj)
-  return push
+function opPush(data: string | Buffer) {
+  const buff = Buffer.isBuffer(data) ? data : Buffer.from(data, "utf8")
+  return Buffer.concat([buff])
 }
```

Comment:
in our opPush, should we enforce that the data is always <= 520? i know chunking will always chunk correctly now but its just a safe guard against any future improvement/refactoring

Reply:
Good point; https://github.com/sadoprotocol/ordit-sdk/pull/80/commits/1fcc4748c633f33a5a076d2edd857e3389a25dcd

Reply:
Reverted in #82